lua/neotest-vstest/init.lua

```diff
@@ -92,17 +109,33 @@ function DotnetNeotestAdapter.is_test_file(file_path)
     logger.debug(
       "neotest-vstest: marking file as non-test file since no client was found: " .. file_path
     )
+    is_test_dir_cache[file_path] = false
     return false
   end
 
+  is_test_dir_cache[file_path] = true
   return true
 end
 
-function DotnetNeotestAdapter.filter_dir(name)
+function DotnetNeotestAdapter.filter_dir(name, rel_path, root)
   if name == "bin" or name == "obj" then
     return false
   end
 
+  -- Filter out directories that are not part of the solution (if there is a solution)
+  local fullpath = vim.fs.joinpath(root, rel_path)
+  local project_dir = vim.fs.root(fullpath, function(path, _)
+    return path:match("%.[cf]sproj$")
+  end)
```

Comment:
I think this would break is the project is nested multiple folders away from the solution, e.g.
```txt
- solution
- code
-- projects
--- myproject
---- myproject.csproj
```

Since `filter_dir` recursively descends into directories only if its parent has not been filtered

Reply:
Hmmm, okay so maybe return true if `project_dir` is nil then? If I understand it correctly, then it should descend into child directories, until it eventually finds a project file, and then potentially filters it away if it's not included in the solution?

Reply:
Yeah I imagine something like that would work for any real word scenario. 
Someone could probably still break this by having leaving dummy project files around their solution, but I think it might be quite a deep rabbit hole to guard against all the ways someone can weirdly nest projects 😅

Reply:
Yeah I agree😄 I fixed it in the latest commit